src/main/java/com/school/project/controller/SubjectController.java

```diff
@@ -0,0 +1,40 @@
+package com.school.project.controller;
+
+import com.school.project.model.entities.Subject;
+import com.school.project.service.SubjectService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping
+public class SubjectController {
+
+    @Autowired
+    SubjectService subjectService;
+
+    @PostMapping("/subject")
+    public void createSubject(@RequestBody Subject subject) {
+        subjectService.createSubject(subject);
+    }
+
+    @GetMapping("/subject/{id}")
+    public ResponseEntity<Subject> getSubjectByID(@PathVariable long id) {
+        return ResponseEntity.ok()
+                .body(subjectService.getSubjectById(id));
+    }
+
+    @GetMapping("/subjects")
+    public List<Subject> getAllSubjects() {
+        return subjectService.getAllSubjects();
+    }
+
+    @PutMapping("/subject/{id}")
+    public ResponseEntity<Subject> updateSubject(@RequestBody Subject subject, @PathVariable Long id) {
```

Comment:
- put doesn't return anything by default
- same as above, no Subject entity in this place